### ISSUE TITLE
Question about UseRelativePaths_Experimental usage

### ISSUE THREAD

**Issue by avudnez** [fastbuild/fastbuild#889]
Is it correct to say that `UseRelativePaths_Experimental` does not work with msvc ? (I understand it's experimental and using it is not recommended)

The unit test only runs with clang, so I changed it to use msvc and it still outputs absolute paths in preprocessed output:

```cpp
#line 1 "C:\\git\\fastbuild\\tmp\\Test\\Object\\CacheUsingRelativePaths\\A\\Code\\File.cpp"

#line 1 "C:\\git\\fastbuild\\tmp\\Test\\Object\\CacheUsingRelativePaths\\A\\Code\\Subdir/Header.h"

inline const char* GetFile()
{
    
    return "FILE_MACRO_START_1(" "C:\\git\\fastbuild\\tmp\\Test\\Object\\CacheUsingRelativePaths\\A\\Code\\Subdir/Header.h" ")FILE_MACRO_END_1";
}
#line 3 "C:\\git\\fastbuild\\tmp\\Test\\Object\\CacheUsingRelativePaths\\A\\Code\\File.cpp"

const char* Function()
{
    
    return "FILE_MACRO_START_2(" "C:\\git\\fastbuild\\tmp\\Test\\Object\\CacheUsingRelativePaths\\A\\Code\\File.cpp" ")FILE_MACRO_END_2";
}

const char* Function2()
{
    return GetFile(); 
}
```

**Comment by ben-craig** [comment]
I don't believe there's a way to get MSVC to output relative paths for `__FILE__`.  What #885 (Allow LightCache to use relative paths) does is make it so that LightCache's hashing / "identity" function use relative paths.  This allows more obj file caching.  It doesn't help with making bit-identical builds independent of absolute path.

**Comment by avudnez** [comment]
Also, shouldn't Fastbuild modify relevant compilation arguments (mainly include paths) from absolute to relative ?
If I understand correctly (and your PR explanation confirms this), it will only modify %1 and %2 as seen here :
https://github.com/fastbuild/fastbuild/blob/9cd65ca296a9c9f7de4d5a14443d1a68b26b685f/Code/Tools/FBuild/FBuildCore/ExeDrivers/Compiler/CompilerDriverBase.cpp#L66-L122

**Comment by ben-craig** [comment]
You could modify other arguments, but you don't really need to, because the user can make them relative on their own.  Making the paths relative can't even be done all the time if you cross drive letters.  Automatic relative pathing also gets weird when dealing with the Windows SDK and the compiler's include path.

**Comment by ffulin** [comment]
UseRelativePaths_Experimental doesn't work with MSVC currently, because even when providing relative paths on the command line, MSVC expands some paths to full path internally sucha that the .obj/.pdb files it creates contain full paths. This makes caching those object files somewhat unsafe, so FASTBuild continues to use full paths as part of the hashing algorithm for caching with MSVC.

While these pathing differences are technically wrong and thus can't be ignored by FASTBuild by default, I think it would be ok for users to opt-in to allow sharing in cases like this if they are ok with the side-effects (possibly different debug paths for example).
